### PR TITLE
feat: Use Echo routing for different DID URLs

### PR DESCRIPTION
**Pull request by Toktar**

### CODE REVIEW

**Review by drgomesp — APPROVED** [review]
Very nice! This was a great clean-up and concise usage of Echo (I left some friendly comments but nothing major).

File: types/errors.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+import "fmt"
+
+type IdentityError struct {
+	Code            int
+	Message         string
+	Internal        error
+	Did             string
+	ContentType     ContentType
+	DispalayMessage func() ResolutionResultI
```
Comment (askolesov) [inline types/errors.go:11]: Why do we use lambda here?

Comment (askolesov) [reply types/errors.go:11]: Can we just add `isDereferencing` property instead?

**Review by askolesov — APPROVED** [review]
Great work!

File: types/errors.go (hunk 1 above)
Comment (Toktar) [reply types/errors.go]: Yes, thank you.
It seems that I tried to make the code as intimidating as possible with its seriousness. 😄 
Fixed

File: cmd/error_handler.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	"github.com/labstack/echo/v4"
+	"github.com/rs/zerolog/log"
+)
+
+func CustomHTTPErrorHandler(err error, c echo.Context) {
+	if err == nil {
+		return
+	}
+	identityError := generateIdentityError(err)
+	if identityError.Code == 500 {
```
[the remark this answers is not in the document]
Comment (Toktar) [reply cmd/error_handler.go]: Thanks. Done

File: cmd/error_handler.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		log.Error().Err(identityError.Internal)
+	} else {
+		log.Warn().Err(identityError.Internal)
+	}
+	c.Response().Header().Set(echo.HeaderContentType, string(identityError.ContentType))
+	err = c.JSONPretty(identityError.Code, identityError.DispalayMessage(), "  ")
```
[the remark this answers is not in the document]
Comment (Toktar) [reply cmd/error_handler.go]: Thanks. Done

File: cmd/error_handler.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+func generateIdentityError(err error) *types.IdentityError {
+	identityError, ok := err.(*types.IdentityError)
+	if ok {
+		return identityError
+	}
+	he, ok := err.(*echo.HTTPError)
+	if !ok || he.Code != 404 {
+		return types.NewInternalError("", types.JSON, err, false)
+	}
+	return types.NewInvalidDIDUrlError("", types.JSON, err, false)
```
[the remark this answers is not in the document]
Comment (Toktar) [reply cmd/error_handler.go:37]: I wanted to make creating errors with empty fields as uncomfortable as possible, as this is not quite normal for a Resolver error.
But I can change it

**Comment by Toktar** [comment]
@drgomesp @askolesov Thanks a lot for your review. 
I processed your comments. Could you please check it?

**Review by drgomesp — APPROVED** [review]

**Review by askolesov — APPROVED** [review]

**Pull request merged**